src/tracking_arc.rs

```diff
@@ -0,0 +1,140 @@
+#[cfg(feature = "internal-track-arc")]
+use crate::id_set::{self, IdSet};
+use std::sync::Arc;
+#[cfg(feature = "internal-track-arc")]
+use std::sync::Mutex as SyncMutex;
+
+/// An `Arc` that records the backtraces of construction of live clones.  When not compiled
+/// with the `internal-track-arc` feature, a zero-cost `Arc` wrapper.
+#[derive(Debug)]
+pub(crate) struct TrackingArc<T> {
```

Comment:
This is largely a relic of the `AsyncResource` approach, but being able to track what `Client` clones are active and where they were created seemed useful enough to preserve.